### PR TITLE
web/satellite: added functionality to remove yourself from someone else's project

### PR DESCRIPTION
**Pull request by VitaliiShpital**
What: now user can delete himself from someone else's project

Why: enhancements

Please describe the tests:
 - Test 1:
 - Test 2:
 
Please describe the performance impact:

## Code Review Checklist (to be filled out by reviewer)
 - [x] NEW: Are there any Satellite database migrations? Are they forwards _and_ backwards compatible? 
 - [x] Does the PR describe what changes are being made?
 - [x] Does the PR describe why the changes are being made?
 - [x] Does the code follow [our style guide](https://github.com/storj/docs/blob/master/code/Style.md)?
 - [x] Does the code follow [our testing guide](https://github.com/storj/docs/blob/master/code/Testing.md)?
 - [x] Is the PR appropriately sized? (If it could be broken into smaller PRs it should be)
 - [x] Does the new code have enough tests? (*every* PR should have tests or justification otherwise. Bug-fix PRs especially)
 - [x] Does the new code have enough documentation that answers "how do I use it?" and "what does it do?"? (both source documentation and [higher level](https://github.com/storj/docs), diagrams?)
 - [x] Does any documentation need updating?
 - [x] Do the database access patterns make sense?

### CODE REVIEW

File: web/satellite/src/store/modules/projects.ts [hunk 1]
```diff
@@ -10,6 +10,7 @@ export const PROJECTS_ACTIONS = {
     SELECT: 'selectProject',
     UPDATE: 'updateProject',
     DELETE: 'deleteProject',
+    DELETE_STORED_PROJECT: 'deleteStoredProject',
```
Comment (crawter) [inline web/satellite/src/store/modules/projects.ts]: DELETE_SELECTED_PROJECT

File: web/satellite/src/components/team/HeaderArea.vue [hunk 2]
```diff
@@ -164,6 +175,35 @@ export default class HeaderArea extends Vue {
...
+            await this.$notify.error(error.message);
+            this.isDeleteClicked = false;
+
+            return;
+        }
+    }
+
+    private async setProjectState(): Promise<void> {
+        const projects = await this.$store.dispatch(PROJECTS_ACTIONS.FETCH);
+        if (projects.length) {
```
Comment (crawter) [inline web/satellite/src/components/team/HeaderArea.vue]: if (!projects.length) {
        await this.$store.dispatch(APP_STATE_ACTIONS.CHANGE_STATE, AppState.LOADED_EMPTY);
        await this.$router.push(RouteConfig.ProjectOverview.with(RouteConfig.ProjectDetails).path);
            
            return;
        }
            await this.$store.dispatch(PROJECTS_ACTIONS.SELECT, projects[0].id);
            await this.$store.dispatch(PM_ACTIONS.FETCH, this.FIRST_PAGE);
            this.$refs.headerComponent.clearSearch();

File: web/satellite/src/components/team/HeaderArea.vue [hunk 3]
```diff
@@ -125,19 +128,27 @@ export default class HeaderArea extends Vue {
     }
 
     public async onDelete(): Promise<void> {
+        const currentUser = this.$store.getters.user;
```
Comment (NikolaiYurchenko) [inline web/satellite/src/components/team/HeaderArea.vue]: remove redundant getters

File: web/satellite/src/components/team/HeaderArea.vue [hunk 4]
```diff
@@ -164,6 +175,35 @@ export default class HeaderArea extends Vue {
...
+        try {
+            await this.$store.dispatch(PM_ACTIONS.DELETE);
+            await this.$store.dispatch(PROJECTS_ACTIONS.DELETE_STORED_PROJECT, selectedProject.id);
+
+            await this.setProjectState();
+        } catch (error) {
+            await this.$notify.error(error.message);
+            this.isDeleteClicked = false;
+
+            return;
```
Comment (NikolaiYurchenko) [inline web/satellite/src/components/team/HeaderArea.vue]: redundant return

File: web/satellite/src/components/team/HeaderArea.vue [hunk 5]
```diff
@@ -127,17 +130,17 @@ export default class HeaderArea extends Vue {
     public async onDelete(): Promise<void> {
         try {
             await this.$store.dispatch(PM_ACTIONS.DELETE);
+            this.setProjectState();
```
Comment (crawter) [inline web/satellite/src/components/team/HeaderArea.vue]: await missing

**Pull request merged**